fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProvider.java

```diff
@@ -34,17 +33,27 @@
 public interface SubspaceProvider {
```

Comment:
What do people think about making this interface either `UNSTABLE` or `INTERNAL`? I think for almost all use cases where someone could theoretically provide a `SubspaceProvider` and is an external client, they can (or should) be able to provide either a `KeySpacePath` or a `Subspace` instead, and then the `SubspaceProvider` is just an algebraic data type we keep for ourselves.

Reply:
+1.  I think with it in the 2.6 branch we should mark it `INTERNAL`.

Reply:
Will mark it INTERNAL after talking with Scott Gray.